src/java/org/apache/nutch/crawl/CrawlDb.java

```diff
@@ -111,7 +116,7 @@ public void update(Path crawlDb, Path[] segments, boolean normalize,
       LOG.info("CrawlDb update: Merging segment data into db.");
     }
     try {
-      JobClient.runJob(job);
+      int complete = job.waitForCompletion(true)?0:1;
     } catch (IOException e) {
```

Comment:
@sebastian-nagel in the above job when there is an IOException the output path is being removed (cleaning up) before throwing up the error and the same needs to be done when we catch the other two exceptions i.e ClassNotFoundException and InterruptedException also, isn't it? Please feel free to correct me if I am wrong.